SocietalConstructionTool/Compiler/SctTranslator.cs

```diff
@@ -174,15 +174,79 @@ public override void ExitWhile([NotNull] SctParser.WhileContext context)
         // WE NEED TO DROP BLOCKS FROM THE STACK UNTILL THEY ARE PROPERLY IMPLEMENTED
         public override void ExitIf([NotNull] SctParser.IfContext context)
         {
-            _ = _stack.Pop();
+            var childBlockNode = _stack.Pop();
```

Comment:
Refactor all this to a single function returning the IfSyntaxDeclaration `@if`, reducing `ExitIf` and `ExitIfElse` to being only a call to that function and pushing to stack, with `ExitIfElse` wrapping in the else block as done currently.

Current implementation repeats a lot of code.